### ISSUE TITLE
Limit fetching to dates of interest

### ISSUE THREAD

**Issue by rfdonnelly** [rfdonnelly/scma-gcal-sync#13]
scma-gcal-sync currently fetches _all_ events when fetching the event list but then only uses the present/active and future events by default (see the `--all` option).  This puts unnecessary load on the SCMA server and creates unnecessary traffic on the wire.

After some searching, I found some source code for Ohanah, the third-party Joomla extension that the SCMA website uses to manage events.  I reviewed the source and found some promising query parameters.  In particular, I found the `starting_date` parameter (see [views/events/json.php:39](https://github.com/Mcfeell/ohanah/blob/6a5736f37dcc4e57f4f31124f34633a533ededea/site/components/com_ohanah/views/events/json.php#L39)).

I tried this with a `YYYY-MM-DD` formatted date and it worked!

Update the event list fetching to include this when the [min_date](https://github.com/rfdonnelly/scma-gcal-sync/blob/993dc87f35240bf599e0ad9fb628ae4001d66177/src/input/web.rs#L18) is `Some(date)`.

**Comment by rfdonnelly** [comment]
On reviewing the code at [views/events/json.php:39](https://github.com/Mcfeell/ohanah/blob/6a5736f37dcc4e57f4f31124f34633a533ededea/site/components/com_ohanah/views/events/json.php#L39), it seems that the `starting_date` query parameter at [models/events.php:68:67](https://github.com/Mcfeell/ohanah/blob/6a5736f37dcc4e57f4f31124f34633a533ededea/site/components/com_ohanah/models/events.php#L68-L69) which matches against the event `end_date`.

However, after testing the `starting_date` query parameter, it seems to match against the event start `date` as shown at [models/events.php:74-75](https://github.com/Mcfeell/ohanah/blob/6a5736f37dcc4e57f4f31124f34633a533ededea/site/components/com_ohanah/models/events.php#L74-L75).

So it seems it is the [model/events.php](https://github.com/Mcfeell/ohanah/blob/6a5736f37dcc4e57f4f31124f34633a533ededea/site/components/com_ohanah/models/events.php) file that should be used to determine the available query parameters and how they filter events.  We want to retrieve all events where the `end_date` is >= the current date.  So we actually want to use the `calendar_start_date` query parameter at [models/events.php:68:67](https://github.com/Mcfeell/ohanah/blob/6a5736f37dcc4e57f4f31124f34633a533ededea/site/components/com_ohanah/models/events.php#L68-L69).  Testing confirms this.

Further review of [model/events.php](https://github.com/Mcfeell/ohanah/blob/6a5736f37dcc4e57f4f31124f34633a533ededea/site/components/com_ohanah/models/events.php) also shows that the `filterEvents=notpast` query will provide all current and future events relative to the SCMA website's time.  In other words, if today is 2022-01-01 then the query strings `filterEvents=notpast` and `calendar_start_date=2022-01-01` are equivalent.

In summary, the two options for implementing this are the query strings:

* `filterEvents=notpast`
* `calendar_start_date=YYYY-MM-DD` where `YYYY-MM-DD` == `today()`
